[aotinductor] Refactor test_aot_inductor to take different devices

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110216
* #110215

Summary: Replace hardcoded device to self.device, to make it easier to test both cpu and cuda

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

desertfire committed:
[aotinductor] Refactor test_aot_inductor

Summary: Replace hardcoded device to self.device, to make it easier to test both cpu and cuda

[ghstack-poisoned]

angelayi commented on test/inductor/test_aot_inductor.py:
```diff
@@ -319,25 +310,25 @@ def test_seq(self):
 
```
Should test_seq also be updated to use self.device for example_inputs?

angelayi commented on test/inductor/test_aot_inductor.py:
```diff
@@ -110,8 +110,10 @@ def check_model(
 ):
-    expected = model(*example_inputs)
-    with config.patch("aot_inductor.abi_compatible", self.abi_compatible):
+    with torch.no_grad(), config.patch(
```
I wonder if it's easier to just have a statement in check_model which converts all the inputs to the specific device using .to instead of initializing all of the tensors with self.device in the tests?

desertfire replied:
That's a good idea, unless there is a mixed of input tensor devices, but I guess we can always add another form of `check_model` in that case. Let me update.

desertfire replied:
Hmm, I see there are tests using constraints constructed with a weak ref to tensor, if we convert input tensors in `check_model`, would that cause a problem? I am seeing any problem with local change, but I am not sure it is safe.

angelayi replied:
Ah, I think you might be right, I checked and the IDs of the tensor from .to is different than the original id...

angelayi commented on test/inductor/test_aot_inductor.py:
```diff
@@ -594,34 +582,86 @@ def forward(self, q, k, v, x):
+    check_model_with_multiple_inputs = check_model_with_multiple_inputs
+
+
+copy_tests(
```
This is cool! First time seeing tests set up like this!

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -61,7 +61,7 @@ def load(cls, model, example_inputs, options=None, constraints=None):
             functions=["run"],
             extra_ldflags=[so_path],
-            with_cuda=True,
+            with_cuda=True,  # TODO: change this to not is_cpu
```
@chenyang78 , you will need to change this to test your runtime logic separation.

bertmaher reviewed (approved):
Nice!

desertfire committed:
Update on "[aotinductor] Refactor test_aot_inductor to take different devices"


Summary: Replace hardcoded device to self.device, to make it easier to test both cpu and cuda

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Refactor test_aot_inductor to take different devices"


Summary: Replace hardcoded device to self.device, to make it easier to test both cpu and cuda

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Refactor test_aot_inductor to take different devices"


Summary: Replace hardcoded device to self.device, to make it easier to test both cpu and cuda

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -h

desertfire commented:
@pytorchbot merge

desertfire committed:
Update on "[aotinductor] Refactor test_aot_inductor to take different devices"


Summary: Replace hardcoded device to self.device, to make it easier to test both cpu and cuda

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge
